wd_helper.au3

```diff
@@ -260,10 +260,11 @@ EndFunc   ;==>_WD_Attach
 ; #FUNCTION# ====================================================================================================================
 ; Name ..........: _WD_LinkClickByText
 ; Description ...: Simulate a mouse click on a link with text matching the provided string.
-; Syntax ........: _WD_LinkClickByText($sSession, $sText[, $bPartial = Default])
-; Parameters ....: $sSession - Session ID from _WD_CreateSession
-;                  $sText    - Text to find in link
-;                  $bPartial - [optional] Search by partial text? Default is True
+; Syntax ........: _WD_LinkClickByText($sSession,  $sText[,  $bPartial = Default[,  $sStartElement = Default]])
+; Parameters ....: $sSession      - Session ID from _WD_CreateSession
+;                  $sText         - Text to find in link
+;                  $bPartial      - [optional] Search by partial text? Default is True
+;                  $sStartElement - [optional] Element ID of element to use as starting point
```

Comment:
Element ID of element (from _WD_FindElement or _WD_WaitElement)

Reply:
other than that one remark, everything else seems okay.

Reply:
I disagree. IMO, it shouldn't be necessary to list these functions by name. Plus there are other functions that return an Element ID, ie: _WD_GetMouseElement, _WD_GetElementFromPoint, _WD_GetShadowRoot, etc.

Reply:
I will review this tomorrow.

Reply:
Ok.
Agree bo all other `Element ID of element` wording should be reviewed, as for example: `_WD_GetShadowRoot` is using such description: `$sStartElement - [optional] a string value. Default is ""`

If you agree please create related issue.

btw.
this PR #396 is fine, as for me.

Reply:
AFAICS, `_WD_GetShadowRoot` is the only function where this needs to be revised.

Reply:
> AFAICS, `_WD_GetShadowRoot` is the only function where this needs to be revised.

also description for return vaule in `_WD_GetElementByRegEx`

Reply:
and maybe `$sStartNodeID - [optional] Element ID to use as starting node. Default is ""` from `_WD_FindElement`

Reply:
also there are few places where description like `Element ID from _WD_FindElement`  is already used.

Reply:
What are you thoughts about changing all instances of `$sStartElement` to `$sStartNodeID` so that the parameter naming is "standardized"?

Reply:
Good idea.
Please take care of this.

Reply:
the same with `$sElement` >> `$sElementNodeID` ?

Reply:
for example 

in `_WD_LoadWait`
current:
`;                  $sElement - [optional] Element ID (from _WD_FindElement or _WD_WaitElement) to confirm DOM invalidation`
proposal:
`;                  $sElementNodeID - [optional] ElementNodeID to confirm DOM invalidation`


in `_WD_FindElement`
current:
`; Return values .: Success - ElementNodeID(s) returned by web driver.`
proposal:
`; Return values .: Success - ElementNodeID(s) returned by web driver.`

Reply:
> the same with $sElement >> $sElementNodeID ?

Not sure about that. This is the description for $sStartNodeID --

`;                  $sStartNodeID - [optional] Element ID to use as starting node. Default is ""`

This make sense to me because the parameter name matches the description (although the word `element` isn't utilitzed).

I don't like the use of `element` and `node` together in the same variable name because they are essentially the same thing IMO. One could argue that `$sElement` >> `$sElementID` would be more consistent naming.

Reply:
> One could argue that `$sElement` >> `$sElementID` would be more consistent naming.

This would be good.

Reply:
But what about such change:

in _WD_FindElement
current:
; Return values .: Success - Element ID(s) returned by web driver.
proposal:
; Return values .: Success - ElementID(s) returned by web driver.

and in `_WD_ElementOptionSelect`
current:
`;                  $sStartElement - [optional] Element ID of element to use as starting point`
proposal:
`;                  $sStartNodeID - [optional] ElementID of element to use as starting point`
or better proposal:
`;                  $sStartElementID - [optional] ElementID of element to use as starting point`

it would stadarize to 
`ElementID` and `$sStartElementID` and `$sElementID`

without spaces `Element ID`

Reply:
btw.
https://stackoverflow.com/a/9979779/5314940
`An element is a specific type of node,`

can `_WD_FIndElement` finds any node types ?
or only `Node.ELEMENT_NODE (1)` https://developer.mozilla.org/en-US/docs/Web/API/Node/nodeType

Reply:
> I don't like the use of `element` and `node` together in the same variable name because they are essentially the same thing IMO.

In this reference documentation:
https://developer.mozilla.org/en-US/docs/Web/API/Node/nodeType

They use such naming 
`Node.ELEMENT_NODE (1)` 

and ... ``Node.ATTRIBUTE_NODE``

so `element` and `node` can be used together  to describe exactly what type of Node is required.